File: apps/mobile/lib/features/seller/models/seller_application.dart
/// Mirrors `services/api/internal/httpapi/dto/seller_application.go`'s
/// SellerApplicationResponse/CreateSellerApplicationRequest. Plain classes
/// (no freezed/json_serializable) — this is a small, one-off shape used
/// only by the become-a-seller wizard, not worth the generated-file
/// overhead the rest of the app's shared models carry.
class SellerApplication {
  const SellerApplication({
    required this.id,
    required this.status,
    required this.birthDate,
    this.storeLat,
    this.storeLng,
    this.storeWebsite,
    this.storeInstagram,
    this.storeTelegram,
    this.storeWhatsapp,
    required this.livenessPassed,
    this.faceMatchScore,
    this.rejectionReason,
    required this.createdAt,
  });

  factory SellerApplication.fromJson(Map<String, dynamic> json) {
    return SellerApplication(
      id: json['id'] as String,
      status: json['status'] as String,
      birthDate: json['birth_date'] as String,
      storeLat: (json['store_lat'] as num?)?.toDouble(),
      storeLng: (json['store_lng'] as num?)?.toDouble(),
      storeWebsite: json['store_website'] as String?,
      storeInstagram: json['store_instagram'] as String?,
      storeTelegram: json['store_telegram'] as String?,
      storeWhatsapp: json['store_whatsapp'] as String?,
      livenessPassed: json['liveness_passed'] as bool? ?? false,
      faceMatchScore: (json['face_match_score'] as num?)?.toDouble(),
      rejectionReason: json['rejection_reason'] as String?,
      createdAt: json['created_at'] as String,
    );
  }

  final String id;
  final String status; // 'pending' | 'approved' | 'rejected'
  final String birthDate;
  final double? storeLat;
  final double? storeLng;
  final String? storeWebsite;
  final String? storeInstagram;
  final String? storeTelegram;
  final String? storeWhatsapp;
  final bool livenessPassed;
  final double? faceMatchScore;
  final String? rejectionReason;
  final String createdAt;

  bool get isPending => status == 'pending';
  bool get isApproved => status == 'approved';
  bool get isRejected => status == 'rejected';
}

/// Request body for `POST /seller-applications`.
class CreateSellerApplicationRequest {
  const CreateSellerApplicationRequest({
    required this.birthDate,
    this.storeLat,
    this.storeLng,
    this.storeWebsite,
    this.storeInstagram,
    this.storeTelegram,
    this.storeWhatsapp,
    required this.passportFrontKey,
    required this.passportBackKey,
    required this.selfieWithPassportKey,
    required this.liveSelfieKey,
    required this.livenessPassed,
    this.faceMatchScore,
  });

  final String birthDate; // "YYYY-MM-DD"
  final double? storeLat;
  final double? storeLng;
  final String? storeWebsite;
  final String? storeInstagram;
  final String? storeTelegram;
  final String? storeWhatsapp;
  final String passportFrontKey;
  final String passportBackKey;
  final String selfieWithPassportKey;
  final String liveSelfieKey;
  final bool livenessPassed;
  final double? faceMatchScore;

  Map<String, dynamic> toJson() => {
        'birth_date': birthDate,
        'store_lat': storeLat,
        'store_lng': storeLng,
        'store_website': storeWebsite,
        'store_instagram': storeInstagram,
        'store_telegram': storeTelegram,
        'store_whatsapp': storeWhatsapp,
        'passport_front_key': passportFrontKey,
        'passport_back_key': passportBackKey,
        'selfie_with_passport_key': selfieWithPassportKey,
        'live_selfie_key': liveSelfieKey,
        'liveness_passed': livenessPassed,
        'face_match_score': faceMatchScore,
      };
}

File: services/api/internal/httpapi/dto/seller_application.go
package dto

import "tajikshop/api/internal/models"

// CreateSellerApplicationRequest is the body for POST /seller-applications.
// StoreLat/StoreLng OR at least one of the Store* contact links must be set
// (validated in service.SellerApplicationService, not here — see
// docs/API_SPEC.md). The *Key fields are object keys returned by
// POST /uploads/presign with purpose="seller-kyc" (never a public URL).
type CreateSellerApplicationRequest struct {
	BirthDate             string   `json:"birth_date" binding:"required"` // "YYYY-MM-DD"
	StoreLat              *float64 `json:"store_lat"`
	StoreLng              *float64 `json:"store_lng"`
	StoreWebsite          *string  `json:"store_website"`
	StoreInstagram        *string  `json:"store_instagram"`
	StoreTelegram         *string  `json:"store_telegram"`
	StoreWhatsapp         *string  `json:"store_whatsapp"`
	PassportFrontKey      string   `json:"passport_front_key" binding:"required"`
	PassportBackKey       string   `json:"passport_back_key" binding:"required"`
	SelfieWithPassportKey string   `json:"selfie_with_passport_key" binding:"required"`
	LiveSelfieKey         string   `json:"live_selfie_key" binding:"required"`
	// LivenessPassed/FaceMatchScore come from the mobile app's on-device
	// liveness challenge + face-similarity check (Google ML Kit face
	// detection/landmarks — free, on-device, no paid KYC API). The server
	// re-checks FaceMatchScore against its own threshold as a second gate,
	// but cannot independently re-run the comparison itself — see
	// docs/SMS_PROVIDERS.md-style caveat in SellerApplicationService's doc
	// comment on the honest limits of a fully free/on-device approach.
	LivenessPassed bool     `json:"liveness_passed"`
	FaceMatchScore *float64 `json:"face_match_score"`
}

// SellerApplicationResponse is returned by POST /seller-applications and
// GET /seller-applications/me.
type SellerApplicationResponse struct {
	ID              string   `json:"id"`
	Status          string   `json:"status"`
	BirthDate       string   `json:"birth_date"`
	StoreLat        *float64 `json:"store_lat"`
	StoreLng        *float64 `json:"store_lng"`
	StoreWebsite    *string  `json:"store_website"`
	StoreInstagram  *string  `json:"store_instagram"`
	StoreTelegram   *string  `json:"store_telegram"`
	StoreWhatsapp   *string  `json:"store_whatsapp"`
	LivenessPassed  bool     `json:"liveness_passed"`
	FaceMatchScore  *float64 `json:"face_match_score"`
	RejectionReason *string  `json:"rejection_reason"`
	CreatedAt       string   `json:"created_at"`
}

// NewSellerApplicationResponse builds the response DTO from a model.
func NewSellerApplicationResponse(a *models.SellerApplication) SellerApplicationResponse {
	return SellerApplicationResponse{
		ID:              a.ID.String(),
		Status:          a.Status,
		BirthDate:       a.BirthDate.Format("2006-01-02"),
		StoreLat:        a.StoreLat,
		StoreLng:        a.StoreLng,
		StoreWebsite:    a.StoreWebsite,
		StoreInstagram:  a.StoreInstagram,
		StoreTelegram:   a.StoreTelegram,
		StoreWhatsapp:   a.StoreWhatsapp,
		LivenessPassed:  a.LivenessPassed,
		FaceMatchScore:  a.FaceMatchScore,
		RejectionReason: a.RejectionReason,
		CreatedAt:       a.CreatedAt.Format("2006-01-02T15:04:05Z07:00"),
	}
}
